From b4b5fbd5a4d46425c73eac070d69c959766d2aea Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Thu, 06 Jul 2023 09:20:38 +0300
Subject: [PATCH] [SW-151130] gaudi2: ppboot: fix bug in clearing bss

The labels used in clearing bss and copying data
section from FW SRAM to DCCM are same and also
lp_count register is overwritten by bss clear
size which in turn copies only 4 bytes from
FWSRAM to ARC0 DCCM. This patch fixes this by
adding new label for bss clear.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: Id444bf3ca03b68035dbd5d50373fee4dad59884a
---

diff --git a/arch/gaudi2/arc/secondary.S b/arch/gaudi2/arc/secondary.S
index f63167a..9679f3c 100644
--- a/arch/gaudi2/arc/secondary.S
+++ b/arch/gaudi2/arc/secondary.S
@@ -30,11 +30,12 @@
 	lpnz    1f
 	ld.ab   r5,[r1,4]
 	st.ab   r5,[r2,4]
+1:
 	/* Clear BSS */
 	mov     r6, __bss_start
 	sub     r7, __bss_end, r6
 	lsr.f   lp_count, r7, 2
-	lpnz    1f
+	lpnz    2f
 	st.ab   0, [r6, 4]
-1:
+2:
 	bl	secondary_main
